testing/test_pipeline.py

```diff
@@ -355,10 +355,12 @@ def test_hyperparam_space():
 
     hyperparams = p.get_hyperparams()
 
-    assert 'AddFeatures__SomeStep1__n_components' in hyperparams.keys()
-    assert 'AddFeatures__SomeStep__n_components' in hyperparams.keys()
-    assert 'AddFeatures__SomeStep1__n_components' in hyperparams.keys()
-    assert 'ModelStacking__SomeStep__n_estimators' in hyperparams.keys()
-    assert 'ModelStacking__SomeStep1__n_estimators' in hyperparams.keys()
-    assert 'ModelStacking__SomeStep2__max_depth' in hyperparams.keys()
-    assert 'ModelStacking__SomeStep3__max_depth' in hyperparams.keys()
+    assert "AddFeatures" in hyperparams.keys()
+    assert "SomeStep" in hyperparams["AddFeatures"]
+    assert "n_components" in hyperparams["AddFeatures"]["SomeStep"]
+    assert "SomeStep1" in hyperparams["AddFeatures"]
+    assert "n_components" in hyperparams["AddFeatures"]["SomeStep1"]
+    assert "SomeStep" in hyperparams["ModelStacking"]
+    assert "n_estimators" in hyperparams["ModelStacking"]["SomeStep"]
+    assert "SomeStep1" in hyperparams["ModelStacking"]
+    assert "max_depth" in hyperparams["ModelStacking"]["SomeStep2"]
```

Comment:
You seem to have changed from flat hyperparams to nested hyperparams. However, your other changes in your present commit [`Make All Steps Have Flatten Hyperparams`](https://github.com/Neuraxio/Neuraxle/pull/51/commits/36363fd3d85b9c5605acfc93fef6b12f3af4b6d5) seems to do the inverse of this. I don't understand how making the hyperparams flat could make this test changed to asserting that they are nested instead of flat. 

Note, there is also an error to fix in this test: replace `rvsed = p.get_hyperparams_space()` just above by `rvsed = p.get_hyperparams_space().rvs()`, because right now a whole space is set as an hyperparam sample but `.rvs()` should be taken to make them samples instead of a space, so the test isn't really testing the good thing.

Reply:
@alexbrillant question: why does it says "outdated" on this comment? This seems impossible to me, because I wrote that while seeing the particular diff of the most recent commit ("Make All Steps Have Flatten Hyperparams"). I don't see how the last commit could be outdated in this PR.

Reply:
I found the error: see my comment below about `.to_flat`.